src/core/model_event_queue.jl

```diff
@@ -13,57 +13,117 @@ struct EventQueueABM{
     S<:SpaceType,
     A<:AbstractAgent,
     C<:ContainerType{A},
-    F,P,W,L,R<:AbstractRNG} <: AgentBasedModel{S}
+    T,F,P,W,L,R<:AbstractRNG} <: AgentBasedModel{S}
+    time::Base.RefValue{Float64}
     agents::C
     space::S
     scheduler::F
     properties::P
     rng::R
     maxid::Base.RefValue{Int64}
+    agents_types::T
     all_events::W
     all_rates::L
     event_queue::PriorityQueue{Event, Float64}
 end
 
+"""
+    EventQueueABM <: AgentBasedModel
+
+A concrete implementation of an [`AgentBasedModel`](@ref) which operates in 
+continuous time, in contrast with the discrete nature of [`StandardABM`](@ref). 
+
+Pairs composed of an event and an agent are scheduled at some particular time, 
+and once the model time reaches that time, the event is executed on the agent.
+
+Events must be passed during initialization of the model with the `all_events` keywords;
+they are arbitrary agent stepping functions analogous to the `agent_step!` function used
+currently in the [`StandardABM`](@ref) implementation. Hence, events have access to the 
+full API of `Agents.jl`.
+
+Events that are occurring are chosen randomly during scheduling, based on the Direct Method
+of the [Gillespie Algorithm](https://en.wikipedia.org/wiki/Gillespie_algorithm). Each event 
+has a possible propensity value and the probability for choosing a particular event is 
+proportional to the propensity. During the model time evolution, two things occur:
+
+- the model time reaches the time of the next scheduled event and the event action 
+  is performed on the paired agent;
+- the scheduler generates a new event utilizing the rates corresponding to the agent type,
+  and schedules it together with the same paired agent;
+
+Even if the scheduler generate automatically new events based on the rates, it should be
+initialized by adding some events at the start of the simulation through the [`add_event!`](@ref) 
+function.
+
+## Keywords
+
+- `all_events`: 
```

Comment:
I think that `all_events` should not be keyword but positional mandatory argument. In contrast to StandardABM, this can't work without `all_events`.